hudi-common/src/main/java/org/apache/hudi/common/model/HoodieBaseFile.java

```diff
@@ -115,16 +119,19 @@ private static String[] handleHudiGeneratedFile(String fileName) {
     }
     // case where there is no '.' in file name (no file suffix like .parquet)
     values[1] = fileName.substring(lastUnderscoreIndex + 1);
+    values[2] = fileName;
     return values;
   }
 
   private static String[] handleExternallyGeneratedFile(String fileName) {
-    String[] values = new String[2];
+    String[] values = new String[3];
     // file name has format <originalFileName>_<commitTime>_hudiext and originalFileName is used as fileId
     int lastUnderscore = fileName.lastIndexOf(UNDERSCORE);
     int secondToLastUnderscore = fileName.lastIndexOf(UNDERSCORE, lastUnderscore - 1);
-    values[0] = fileName.substring(0, secondToLastUnderscore);
+    int firstUnderscore = fileName.indexOf(UNDERSCORE);
```

Comment:
we can't go w/ first underscore. hudi's file name will inherently contain underscores.